PIL/Image.py

```diff
@@ -270,6 +285,7 @@ def isImageType(t):
 
 
 def _conv_type_shape(im):
+    # type: (Image) -> Tuple[Size, Tuple[Text, Optional[int]]]  # NOTE: Size inverted compared to normal
```

Comment:
Yes, I worked on this last night and just committed my version to the PR, with the same annotation as you, though I see you changed the code too.

This appears to be resolved through simply using:
```
    typ, extra = _MODE_CONV[im.mode]
    if extra is None:
        return (im.size[1], im.size[0]), typ
    else:
        return (im.size[1], im.size[0], extra), typ
```
I assume this is since mypy cannot cope with tuple addition (or that it's not as clear-cut as it appears). That said, this looks cleaner to me, in terms of tuple structure, and does remove the need for the shape variable.